### PR TITLE
kasm VNC

### LINKED ISSUE #1: kasmVNC Module
Create a module that installs kasmvnc if it doesn't exist and starts it. 

## Must have

- [x] coder_app
- [x] coder_script (https://github.com/coder/coder/pull/9584)

## Nice to have

- [x] Variable to specify kasmvnc version
- [ ] Variable to skip install (assumes it's already in the image)
- [ ] Variable for path of kasmvnc (e.g. in home dir?)

## References

- A generic script I made to install kasmVNC on an Ubuntu Focal VM: https://gist.github.com/bpmct/4e51d7092089f278c91f53e344749b6c
- KasmVNC Kubernetes templates: https://github.com/sharkymark/v2-templates

### PR DESCRIPTION
**Pull request by matifali**
closes #1 

For testing,
```tf
module "kasmvnc" {
  source   = "git::https://github.com/coder/modules.git//kasmvnc?ref=kasmVNC"
  agent_id = coder_agent.dev.id
}
```
image

### CODE REVIEW

**Comment by matifali** [comment]
@johnstcn, should we publish an image `codercom/enterprise-desktop` with `lxde` pre-installed so that users can base their images on `codercom/enterprise-desktop`.
What do you think?

**Comment by matifali** [comment]
kasmVNC has multi virtual monitor support

https://github.com/coder/modules/assets/10648092/6a21cfa2-57a7-4b09-9563-fc511faa084f

**Comment by matifali** [comment]
@code-asher, could you check why these unrelated tests are failing?

**Comment by code-asher** [comment]
Could be this maybe, it came out in the release 5 days ago and seems related:  https://github.com/coder/terraform-provider-coder/commit/8c5e8ff9579d1d766d5317422bf3c1988b710434

One test, for example, expects there to be no committer email, but there is one which is `default@example.com`.

So I suppose how we fix depends on whether we want these to be empty or if we should be expecting defaults.

**Comment by michaelbrewer** [comment]
@matifali come comments for how this could be enterprise friendly.

- Ubuntu as the base image is not hardened enough and lack enterprise support/sla, and something like Red Hat Enterprise Linux 9 would be more suitable. So `dnf`, `rpm` and `yum` over the use of `apt` and `dpkg`.

- It might be simpler / faster to leave the installation part to the base image (or VM), and let the script just verify the presence. 

- The signaling when ready pattern ie: `/tmp/.coder-${WAIT_FOR_SCRIPT}.done`, could be done more generically across modules, ie: git-clone waiting on git-config, jetbrains waiting on git-clone etc..

**Comment by matifali** [comment]
@michaelbrewer Thanks for the review

> It might be simpler / faster to leave the installation part to the base image (or VM), and let the script just verify the presence.

Thats the plan.

> The signaling when ready pattern ie: /tmp/.coder-${WAIT_FOR_SCRIPT}.done, could be done more generically across modules, ie: git-clone waiting on git-config, jetbrains waiting on git-clone etc..

It was only done for testing purposes with our dogfood template.

I plan to come to this once I find some time. The idea is to leave the DE installation to image and only check for compatible ones. Also, I will make the script general enough to be installable on all enterprise-friendly distros.

File: kasmvnc/main.tf [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  type        = string
+  description = "Version of KasmVNC to install."
+  default     = "1.3.1"
+}
+
+variable "desktop_environment" {
+  type        = string
+  description = "Specifies the desktop environment of the workspace. This should be pre-installed on the workspace."
+  validation {
+    condition     = contains(["xfce", "kde", "gnome", "lxde", "lxqt", "xfce"], var.desktop_environment)
```
Comment (matifali) [inline kasmvnc/main.tf:33]: as shown in https://github.com/kasmtech/KasmVNC/blob/e5404b10e159c49229c6854f421fb3022bbd7c30/builder/startup/deb/select-de.sh#L75

**Comment by phorcys420** [comment]
@matifali 

Here's what happens when I install the KasmVNC module in my [uwu/basic-env](https://github.com/uwu/basic-env) template.
I think it's due to the fact that Kasm thinks `vncserver` being installed means KasmVNC is.

```
Detected Distribution: ubuntu
Detected Version: 24.04
Detected Architecture: x86_64
vncserver is already installed.
vncserver already installed. Skipping installation.
bash: line 1: /etc/kasmvnc/kasmvnc.yaml: No such file or directory
🚀 Starting KasmVNC server...
Usage: vncpasswd [FILE]
       vncpasswd -t
```

I will retry with a template that doesn't have VNC but it looks good to me other than that.

**Review by phorcys420 — APPROVED** [review]
LGTM!

File: kasmvnc/run.sh [hunk 2]
```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (phorcys420) [reply kasmvnc/run.sh:1]: still have that question pending but it's not a blocker :-)

[the remark this answers is not in the document]
Comment (matifali) [reply kasmvnc/run.sh:1]: They don't have one. Do they?
While writing this I felt I am writing an install script for them.

**Review by jatcod3r — APPROVED** [review]
looks good to me.

**Comment by matifali** [comment]
I still need to update the READEME to mention our desktop image. Then I will merge.

**Pull request merged**